src/utils.hpp

```diff
@@ -0,0 +1,13 @@
+#pragma once
+
+#include <memory>
+
+namespace cose {
+
+template <typename T>
+inline std::shared_ptr<T> make_managed_array(size_t size)
+{
+	return std::shared_ptr<T>(new T[size], [](T *p) { delete[] p; });
+}
+
+}  // namespace cose
```

Comment:
this file is just an example of how we could improve the overall code of cose c

* add namespace to prevent collisions
* add more managed mamory stuff e.g. unique pointers and shared pointers with automatic distruction

Reply:
so
~~~
template <typename T>
inline std::shared_ptr<t> make_managed_array(size_t size)
{
         return std::shared_ptr<T>((COSE_ALLOC(1, size, context)), [](T * p) { COSE_FREE( p, context); )
}
~~~~~

Reply:
there is no `COSE_ALLOC` so far